pkg/operator/injection/injection.go

```diff
@@ -68,15 +53,23 @@ func GetControllerName(ctx context.Context) string {
 // dynamically at runtime due to the necessity of having to build logic around re-queueing to ensure that settings are
 // properly reloaded for things like feature gates
 func WithSettingsOrDie(ctx context.Context, kubernetesInterface kubernetes.Interface, settings ...settings.Injectable) context.Context {
-	cancelCtx, cancel := context.WithCancel(ctx)
+	cancelCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
 	defer cancel()
 
 	factory := informers.NewSharedInformerFactoryWithOptions(kubernetesInterface, time.Second*30, informers.WithNamespace(system.Namespace()))
 	informer := factory.Core().V1().ConfigMaps().Informer()
 	factory.Start(cancelCtx.Done())
 
 	for _, setting := range settings {
-		cm := lo.Must(WaitForConfigMap(ctx, setting.ConfigMap(), informer))
+		cm, err := WaitForConfigMap(ctx, setting.ConfigMap(), informer)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				panic(err)
```

Comment:
Should we provide any additional context to this error?

Reply:
Also, this should be `if !errors.IsNotFound(err)` since that's the only error type that we want to ignore here

Reply:
Yep, that's a miss. It doesn't look like `WaitForConfigMap` provides much context so I think it makes sense to add some here (e.g. `panic(fmt.Errorf("failed to get configmap, %w", err)`).